lib/crawly/response.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Crawly.Response do
+  @moduledoc """
+  Define Crawly response structure
+  """
+
+  defstruct body: nil,
+            headers: [],
+            request: nil,
+            request_url: nil,
+            status_code: nil
+
+  @type t :: %__MODULE__{
+               body: term(),
+               headers: list(),
+               request: Crawly.Request.t(),
+               request_url: Crawly.Request.url(),
```

Comment:
Is the request_url field necessary? It is already in the Request map. For convenience?

Reply:
I think it's driven by HTTPoison. My understanding that request url might be different. For example in the case of redirect:
```
iex(1)> HTTPoison.get("http://meta.ua")
{:ok,
 %HTTPoison.Response{
   body: "<html>\r\n<head><title>301 Moved Permanently</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>301 Moved Permanently</h1></center>\r\n<hr><center>nginx/1.14.0</center>\r\n</body>\r\n</html>\r\n",
   headers: [
     {"Server", "nginx/1.14.0"},
     {"Date", "Mon, 30 Dec 2019 13:00:44 GMT"},
     {"Content-Type", "text/html"},
     {"Content-Length", "185"},
     {"Connection", "keep-alive"},
     {"Location", "https://meta.ua/"}
   ],
   request: %HTTPoison.Request{
     body: "",
     headers: [],
     method: :get,
     options: [],
     params: %{},
     url: "http://meta.ua"
   },
   request_url: "http://meta.ua",
   status_code: 301
 }}```

Reply:
I did some digging, turns out that HTTPoison mimic's python's request library, such that the `request_url` key reflects the final requested url made, after applying modifications such as params.

relevant HTTPoison issue:
https://github.com/edgurgel/httpoison/issues/270

relevant python library:
https://2.python-requests.org//en/master/user/quickstart/#passing-parameters-in-urls

I think it will be good to add in docs on what request_url is used for in the `Crawly.Response` struct, as it will also apply to how Fetchers should output the Response.